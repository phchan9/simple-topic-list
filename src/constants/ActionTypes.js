export const ADD_TOPIC='ADD_TOPIC';
export const UPVOTE_TOPIC='UPVOTE_TOPIC';
export const DOWNVOTE_TOPIC='DOWNVOTE_TOPIC';
