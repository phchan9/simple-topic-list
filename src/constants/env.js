let API_URL = 'http://localhost:3001/api/topics';

if (process.env.NODE_ENV === 'production') {
  API_URL = 'http://api-topics.herokuapp.com/api/topics';
}

export default API_URL;
