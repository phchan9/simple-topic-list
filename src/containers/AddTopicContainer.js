import { connect } from 'react-redux';
import { addTopic } from '../actions';
import AddTopic from '../components/AddTopic';

const mapDispatchToProps = {
  addTopic
};

export default connect(null, mapDispatchToProps)(AddTopic);
