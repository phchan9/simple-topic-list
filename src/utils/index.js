const createTopic = (id, upVotes, downVotes, title) => ({
  id,
  upVotes,
  downVotes,
  title,
});

export default createTopic;
