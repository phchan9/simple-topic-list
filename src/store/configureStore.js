import { createStore, applyMiddleware } from 'redux';
import promise from 'redux-promise';
import { v4 } from 'uuid';
import reducers from '../reducers';
import { createTopic } from '../utils';

const topics = [
  createTopic(v4(), 12, 23, 'Things you should know before using AWSâ€™s Elasticsearch Service'),
  createTopic(v4(), 33, 11,
              'Cannabis reverses aging processes in the brain'),
  createTopic(v4(), 41, 2,
              'Ask HN: How secure is the encryption offered by OS X\'s Disk Utility?'),
  createTopic(v4(), 8, 3,
              'The world\'s largest library of historical European martial arts books')
];

const initialState = {
  byId: topics.reduce((acc, topic) => ({...acc, [topic.id]: topic}), {}),
  allIds: topics.map(topic => topic.id)
};

const store = createStore(reducers, initialState, applyMiddleware(promise));
export default store;
