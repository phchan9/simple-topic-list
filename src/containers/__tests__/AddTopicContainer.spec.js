import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import AddTopicContainer from '../AddTopicContainer';
import { createTopic } from '../../utils';
import * as types from '../../constants/ActionTypes';

jest.mock('node-uuid', () => {
  return {
    v4: jest.fn(() => 100)
  };
});

const mockState = {
  byId:
  {
    1: createTopic(1, 10, 20, 'test1'),
    2: createTopic(2, 33, 22, 'test2'),
    3: createTopic(3, 2, 13, 'test3'),
    4: createTopic(4, 55, 23, 'test4')
  },
  allIds: [1, 2, 3, 4]
};

const setup = () => {
  const mockStore = configureStore()(mockState);
  const wrapper = shallow(<AddTopicContainer store={mockStore} />);

  return {
    mockStore,
    wrapper
  };
};

describe('AddTopicContainer', () => {
  it('render correctly', () => {
    const { wrapper, mockStore } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should show an error that indicate the topic exceed 255 characters', () => {
  });
  const { wrapper } = setup();
  const errorMsgLabelWrapper = wrapper.find('.error').first();
  const form = wrapper.find('form').first();
  form.simulate('submit', {
    target: {
      value: 'a'.repeat(256)
    }
  });
  expect(errorMsgLabelWrapper.hasClass('invisible')).toBe(false);
});
