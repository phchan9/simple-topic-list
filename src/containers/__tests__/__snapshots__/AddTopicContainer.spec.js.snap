// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTopicContainer render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <AddTopicContainer
    dispatch={[Function]}
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  "nodes": Array [
    <AddTopicContainer
      dispatch={[Function]}
      store={
            Object {
                  "clearActions": [Function],
                  "dispatch": [Function],
                  "getActions": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                }
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Connect(AddTopicContainer)
        store={
                Object {
                        "clearActions": [Function],
                        "dispatch": [Function],
                        "getActions": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                      }
        }
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Connect {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "store": undefined,
          "storeSubscription": undefined,
        },
        "props": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "propsMode": true,
        "refs": Object {},
        "renderCount": 0,
        "selector": Object {
          "error": null,
          "props": Object {
            "dispatch": [Function],
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
          },
          "run": [Function],
          "shouldComponentUpdate": false,
        },
        "setWrappedInstance": [Function],
        "state": Object {},
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
        "version": 0,
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <AddTopicContainer
          dispatch={[Function]}
          store={
                    Object {
                              "clearActions": [Function],
                              "dispatch": [Function],
                              "getActions": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                            }
          }
/>,
        "_debugID": 4,
        "_renderedOutput": <AddTopicContainer
          dispatch={[Function]}
          store={
                    Object {
                              "clearActions": [Function],
                              "dispatch": [Function],
                              "getActions": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                            }
          }
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Connect(AddTopicContainer)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
}
`;
